### PR TITLE
[docs] Add instructions to run a logging pipeline locally with docker.

### PR DESCRIPTION
**Pull request by naseemkullah**
It is not uncommon to want to see how a logging pipeline works locally, this can be achieved with docker.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [N/A] Example configuration file for the change
- [N/A ] Debug log output from testing the change

- [N/A ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

**Documentation**

- [x] Documentation required for this feature

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -4,19 +4,22 @@ Assuming you have some basic knowledge of C, this guide should help you understa
...
+  - [Message Pack](#message-pack)
 - [Concurrency](#concurrency)
 - [Plugin API](#plugin-api)
-    - [Input](#input)
-    - [Filter](#filter)
-    - [Output](#output)
+  - [Input](#input)
+  - [Filter](#filter)
+  - [Output](#output)
+- [Running Locally](#running)
```
Comment (PettitWesley) [inline DEVELOPER_GUIDE.md:19]: The link needs to be `#running locally`

**Comment by PettitWesley** [comment]
@naseemkullah How does this help Developers of Fluent Bit?

Testing a logging pipeline locally feels like it fits more in a beginners guide to *using Fluent Bit* not in a beginners guide to *contributing to Fluent Bit*.

Let me know what you think!

**Comment by naseemkullah** [comment]
Thanks @PettitWesley,
Where do you suggest this documentation be placed? I was not sure and asked @edsiper but had not gotten a response yet.

It is true that it is geared towards operators that require validating the pipeline's behaviour locally (e.g. the bug in #2093 was caught and made reproducible by this technique). FWIW I imagine one can develop by re-building the container rather than setting up their local to run C, which apparently is a handful on the latest version of MacOS.

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (naseemkullah) [reply DEVELOPER_GUIDE.md]: Indeed, thanks.

**Comment by naseemkullah** [comment]
Hi @PettitWesley, perhaps this is better suited in the docs repo: I've created https://github.com/fluent/fluent-bit-docs/pull/288

**Comment by PettitWesley** [comment]
@naseemkullah Yeah, I think this is better for the docs; closing this one.

**Pull request closed without merging**